### PR TITLE
[fix bug 1546165] Add optional heading/icon to snippets:

### PR DESCRIPTION
**Pull request by jpetto**
- FXASignupSnippet
- NewsletterSnippet
- SendToDeviceSnippet

Update link to unit testing guide in docs.

https://bugzilla.mozilla.org/show_bug.cgi?id=1546165

### CODE REVIEW

**Comment by jpetto** [comment]
I have been defeated by the prettier error in Travis. I don't see anything obvious on that line. FWIW, running `npm run lint:eslint` locally hangs.

**Comment by pdehaan** [comment]
> I have been defeated by the prettier error in Travis. I don't see anything obvious on that line. FWIW, running `npm run lint:eslint` locally hangs.

Which version of node/npm are you using? I checked out the PR and cleared my ./node_modules and ran `npm i` => `npm run lint:eslint` locally and it completed successfully with the 1 error.

```sh
node --version # v10.16.3
npm --version # 6.9.0
```

```diff
# npm run lint:eslint -- --fix

diff --git a/lib/SnippetsTestMessageProvider.jsm b/lib/SnippetsTestMessageProvider.jsm
index 0ed94a3d..e9e562b8 100644
--- a/lib/SnippetsTestMessageProvider.jsm
+++ b/lib/SnippetsTestMessageProvider.jsm
@@ -197,7 +197,8 @@ const MESSAGES = () => [
       scene1_button_background_color: "#ff9400",
       scene1_section_title_icon:
         "resource://activity-stream/data/content/assets/glyph-pocket-16.svg",
-      scene1_section_title_text: "All the Firefox news that's fit to Firefox print!",
+      scene1_section_title_text:
+        "All the Firefox news that's fit to Firefox print!",
       scene2_title: "Let's do this!",
       locale: "en-CA",
       scene2_dismiss_button_text: "Dismiss",
```

**Comment by jpetto** [comment]
...and it's just a line length issue. 🤦‍♂ 

i'm using npm 6.9.0. i _was_ using node 12.4.0, but after your comment tried 10.0.0 (which i already had installed via nvm) with the same results. installed node 10.16.3, but still nothing. i wiped out `node_modules` between each attempt. i get the following each time, and my machine hangs and the fans get to spinning:

```
(jpetto@chimozbook.local)-(09:22 AM Wed Sep 04)
(~/www/mozilla/activity-stream) $ node --version
v10.16.3

(jpetto@chimozbook.local)-(09:23 AM Wed Sep 04)
(~/www/mozilla/activity-stream) $ npm --version
6.9.0

(jpetto@chimozbook.local)-(09:23 AM Wed Sep 04)
(~/www/mozilla/activity-stream) $ npm run lint:eslint

> activity-streams@1.14.3 lint:eslint /Users/jpetto/www/mozilla/activity-stream
> eslint --cache --ext=.js,.jsm,.jsx .
```

fwiw, `npm run tddmc` works fine.

**Comment by piatra** [comment]
@jpetto the project only supports [node 8.* and npm 6.9](https://github.com/mozilla/activity-stream/blob/60966e5505124f3cef2b4e32e4ab504882a4be40/package.json#L77-L81).

File: content-src/asrouter/components/ConditionalWrapper/ConditionalWrapper.jsx [hunk 1]
```diff
@@ -3,5 +3,7 @@
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 // lifted from https://gist.github.com/kitze/23d82bb9eb0baabfd03a6a720b1d637f
-export const ConditionalWrapper = ({ condition, wrap, children }) =>
+const ConditionalWrapper = ({ condition, wrap, children }) =>
```
Comment (piatra) [inline content-src/asrouter/components/ConditionalWrapper/ConditionalWrapper.jsx]: Why was this change needed?

File: contributing.md [hunk 2]
```diff
@@ -52,7 +52,7 @@ Check out [this guide](docs/v2-system-addon/1.GETTING_STARTED.md) on how to inst
 You have identified the bug, written code and now want to get it into the main repo using a [Pull Request](https://help.github.com/articles/about-pull-requests/).
 
 All code is added using a pull request against the `master` branch of our repo.  Before submitting a PR, please go through this checklist:
-- all [unit tests](#unit-tests) must pass
+- all [unit tests](docs/v2-system-addon/unit_testing_guide.md) must pass
```
Comment (piatra) [inline contributing.md]: Thanks!

File: content-src/asrouter/templates/FXASignupSnippet/FXASignupSnippet.schema.json [hunk 3]
```diff
@@ -31,6 +31,24 @@
         {"description": "Main body text of snippet. HTML subset allowed: i, b, u, strong, em, br"}
       ]
     },
+    "scene1_section_title_icon": {
```
Comment (piatra) [inline content-src/asrouter/templates/FXASignupSnippet/FXASignupSnippet.schema.json]: Can you please bump the schema version as well?

File: content-src/asrouter/components/ConditionalWrapper/ConditionalWrapper.jsx (hunk 1 above)
Comment (jpetto) [reply content-src/asrouter/components/ConditionalWrapper/ConditionalWrapper.jsx]: it wasn't technically needed, but since it's a single-serve component it seemed simpler to move from a named export to a default. happy to revert if this will cause confusion.

Comment (piatra) [reply content-src/asrouter/components/ConditionalWrapper/ConditionalWrapper.jsx]: No need makes sense to me :+1:

**Comment by piatra** [comment]
@jpetto This looks good to me, one small issue I noticed is
image
When hovering the snippet the close button is somewhat close to the action button, "looks" broken.
